flickfinder-backend/src/main/java/com/flickfinder/flickfinderbackend/services/MovieFilterService.java

```diff
@@ -0,0 +1,49 @@
+package com.flickfinder.flickfinderbackend.services;
+
+import com.flickfinder.flickfinderbackend.models.Movie;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Flux;
+
+@Service
+public class MovieFilterService {
+
+
+    private final WebClient webClient;
+    private final String APIKEY;
+
+
+
+    public MovieFilterService(WebClient.Builder webClientBuilder, ApiKeyService apiKeyService) {
+        this.webClient = webClientBuilder.baseUrl("https://api.themoviedb.org/3").build();
+        this.APIKEY = apiKeyService.getApiKey();
+    }
+
+    public Flux<Movie> getMoviesByGenre(String genre) {
+        return webClient.get()
+                .uri("/discover/movie?with_genres={genre}&api_key={apiKey}", genre, APIKEY)
+                .retrieve()
+                .bodyToFlux(Movie.class);
```

Comment:
You might consider just getting the movie id from the response and returning an array of movie ids. Then you can put that array into the moviesServies.getMovieDetails function to get the details of each movie to then return to the user.